openedx/core/djangoapps/course_apps/api.py

```diff
@@ -22,8 +23,12 @@ def is_course_app_enabled(course_key: CourseKey, app_id: str) -> bool:
     Returns:
         True or False depending on if the course app is enabled or not.
     """
-    course_app = CourseAppsPluginManager.get_plugin(app_id)
-    is_enabled = course_app.is_enabled(course_key)
+    try:
+        course_app_status = CourseAppStatus.objects.get(course_key=course_key, app_id=app_id)
+        is_enabled = course_app_status.enabled
+    except CourseAppStatus.DoesNotExist:
+        course_app = CourseAppsPluginManager.get_plugin(app_id)
+        is_enabled = course_app.is_enabled(course_key)
```

Comment:
The backfilling is happening on course publish. There is a signal handler for course publish that will go through all the available course apps and enable/disable them. 

The idea is that for new courses this will mean each save will make us recheck the status. For existing courses, they will backfill if the course is ever edited and published, I think changing any setting here will cause them to be republished. 

I can optionally create a migration to do this as a one-time job if that makes sense. 

If the course updates one app, and that app is saving the data to the modulestore, then that will result in a course publish action and that will automatically cache all other apps as well.

Reply:
What if the course (some existing course is not republished)? In that case the cache would never be filled? I think it would be better if backfill the data here as well. And the publish signal can only handle updating the cache if app status was updated from Studio or some other source. The CourseAppStatus is already updated when the course app status is changed via the MFE.

Reply:
The reason I wasn't too worried about backfilling is that if this PR is deployed now, it will start filling in this table for any course that is created, or recent courses that are updated. Meanwhile, this UI will still be hidden. So by the time this UI is actually usable by most course authors, most courses they need will already be cached.

If we do backfill we still need to have some limits in place. A migration that updates all courses could be too much right? Since there could be thousands of courses. 

Should the backfill migration just run for courses that are currently active?